tests/chainer_tests/functions_tests/noise_tests/test_gaussian.py

```diff
@@ -90,4 +90,35 @@ def test_double_backward_gpu(self):
                                    cuda.to_gpu(self.ggv))
 
 
+@testing.parameterize(*testing.product({
+    'specify_eps': [True, False],
+}))
+class TestGaussianEps(unittest.TestCase):
+
+    def setUp(self):
+        self.m = numpy.random.uniform(-1, 1, (3, 2)).astype(numpy.float32)
+        self.v = numpy.random.uniform(-1, 1, (3, 2)).astype(numpy.float32)
+        self.eps = numpy.random.uniform(-1, 1, (3, 2)).astype(numpy.float32)
+
+    def _check(self):
+        eps = self.eps if self.specify_eps else None
+        out, out_eps = functions.gaussian(
+            self.m, self.v, eps=eps, return_eps=True)
+        assert isinstance(out_eps, type(out.array))
+        if eps is None:
+            assert out_eps.shape == out.array.shape
+        else:
+            assert out_eps is eps
```

Comment:
How about checking verifying the returned eps? Sorry, feeling like I'm repeating myself a lot.

Reply:
Hmm, how can I test randomly generated eps?

Reply:
I added test to call `functions.gaussan` again with `out_eps` and compare results.